docker-compose.yml

```diff
@@ -13,5 +13,47 @@ services:
     image: redis:6-alpine
     ports:
       - "127.0.0.1:6379:6379"
+
+  # Optional containers for development
+  install-deps:
+    profiles: [ dev ]
+    image: dev
+    build: .
+    command: bash -c "asdf install; npm install -g yarn; bundle install; yarn"
+    working_dir: /app
+    volumes:
+      - .:/app
+      - node-modules:/app/node_modules
+      - asdf:/root/.asdf
+
+  dev:
```

Comment:
why does it know that this container should not start when you run `docker compose up`?

Reply:
Because it belongs to the "dev" profile.

You will have to run `docker compose --profile dev up`. E.g. add the specific profile to the call.

Reply:
Cool. I did not know this compose feature